code orange - ui

renenel opened the pull request:
<img width="526" alt="Screen Shot 2020-03-15 at 23 42 14" src="https://user-images.githubusercontent.com/13927275/76711350-f9bc5700-6717-11ea-840d-604689312a5a.png">

Commit:
code orange - ui

danageva commented on code-orange/ViewController.swift:
```diff
@@ -9,17 +9,50 @@
 import GoogleMaps
 
-protocol CurrentLocationProvider: class {
+protocol LocationProvider: class {
```
@renenel can we move the protocol to a different file in the model layer?

renenel replied:
done!

danageva commented on code-orange/ViewController.swift:
```diff
@@ -9,17 +9,50 @@
   var currentLocation: CLLocationCoordinate2D { get }
 }
 
+protocol CoordinatesProviderDelegate: class {
```
same here

renenel replied:
This one belongs with the next one - moved both to a new file!

danageva commented on code-orange/ViewController.swift:
```diff
@@ -9,17 +9,50 @@
+  func coordinatesProviderUpdated(_ sender: CoordinatesProvider)
+}
+
+protocol CoordinatesProvider: class {
```
same here

renenel replied:
done! (with the one above)

danageva commented on code-orange/ViewController.swift:
```diff
@@ -9,17 +9,50 @@
+  var delegate: CoordinatesProviderDelegate? { get set }
+}
+
 class ViewController: UIViewController {
```
Can we give this task a meaningful name?

renenel replied:
Done 🚀

danageva reviewed (changes requested):
Very cool, few comments -- please review.

Commit:
Merge branch 'master' of github.com:getnexar/code-orange-ios into ui

Commit:
code orange - ui - CR fixes

Commit:
Merge branch 'master' of github.com:getnexar/code-orange-ios into ui

# Conflicts:
#	code-orange.xcodeproj/project.pbxproj

Commit:
integrate with LocationsProvider and LocationMatcher

- Make logic classes injection based instead of static
- Minor fixes to logic classes
- Integration with logic classes

Commit:
bug fix in LocationsProvider + ran mocking tests

Commit:
ui - annotations

Commit:
code orange - ui - global location provider and minor cleanup

shaniberacha commented on code-orange/MainViewController.swift:
```diff
@@ -140,20 +142,22 @@ class MainViewController: UIViewController {
     view.addSubview(mainStack)
     mainStack.pin(to: view, anchors: [.leading(0), .trailing(0), .top(28), .bottom(0)])
-    loadLocations()
+    locations = locationsProvider?.getLocations()
```
lets make this into a func "getFreshLocations" 
I think it will be a really useful function

renenel replied:
agreed!
